prefixfilter.go

```diff
@@ -0,0 +1,48 @@
+package complete
+
+import (
+	"strings"
+)
+
+//PrefixFilter filters a predictor's options based on the prefix
+type PrefixFilter interface {
+	FilterPrefix(str, prefix string) bool
+}
+
+//PrefixFilteringPredictor is a Predictor that also implements PrefixFilter
+type PrefixFilteringPredictor struct {
+	Predictor        Predictor
+	PrefixFilterFunc func(s, prefix string) bool
+}
+
+func (p *PrefixFilteringPredictor) Predict(a Args) []string {
+	if p.Predictor == nil {
+		return []string{}
+	}
+	return p.Predictor.Predict(a)
+}
+
+func (p *PrefixFilteringPredictor) FilterPrefix(str, prefix string) bool {
+	if p.PrefixFilterFunc == nil {
+		return DefaultPrefixFilter(str, prefix)
+	}
+	return p.PrefixFilterFunc(str, prefix)
+}
+
+//DefaultPrefixFilter is the PrefixFilter used when none is set
+func DefaultPrefixFilter(s, prefix string) bool {
+	return strings.HasPrefix(s, prefix)
+}
+
+//PermissivePrefixFilter always returns true
+func PermissivePrefixFilter(_, _ string) bool {
+	return true
+}
+
+//CaseInsensitivePrefixFilter ignores case differences between the prefix and tested string
+func CaseInsensitivePrefixFilter(s, prefix string) bool {
```

Comment:
Suggestion: Consider a different API where this is a factory method that accepts a `Predictor` and returns a `MatchPredictor`: it attributes it with case insensitivity.
WDYT?

Reply:
I think I like that suggestion if I understand it correctly.  I'm assuming that `MatchPredictor` would be an interface that includes both `Predictor` and `PrefixFilter` (that brings up a naming question which I'm ignoring in this comment to stay focused on this part).

I'll work on an API change that looks like that.

Reply:
I think you understood correctly.
This will also enables us dropping `PrefixFilteringPredictor`.